### ISSUE TITLE
 I want to extend the existing Link dialog form

### ISSUE THREAD

**Issue by zhangY94** [okijhhyu/element-tiptap-vue3#33]
作者你好，我有个问题想要请教一下。
我现在想要写一个自定义的"Link"标签，我想基于项目中的Link弹框进行拓展，例如我想在menu里面点击就支持Link 的text、href的编辑
以及想增加一个下拉框，实现拼接参数在Link的href链接后面
我该如何自定义这个插件，项目里面我并没有看到相关的例子。
总结来说，我想拓展原有的Link弹窗表单

Hello, author. I have a question I'd like to ask.
I want to create a custom "Link" tag now, and I'd like to extend it based on the project's Link dialog. For example, I want to support editing the text and href of the Link by clicking on it in the menu. Additionally, I'd like to add a dropdown menu to append parameters to the href link of the Link.
How can I customize this plugin? I haven't found any relevant examples in the project.
In summary, I want to extend the existing Link dialog form.

**Comment by zhangY94** [comment]
![1694421316698](https://github.com/okijhhyu/element-tiptap-vue3/assets/54827188/201c889c-d009-4430-8d55-cbf192b38ce5)
和这张图片弹框一样的Link 编辑表单

**Comment by zhangY94** [comment]
![image](https://github.com/okijhhyu/element-tiptap-vue3/assets/54827188/169dede2-311b-44fe-9eab-91ed67ffc175)
我想要实现的Link弹窗

**Comment by okijhhyu** [comment]
![Снимок экрана от 2023-09-12 16-56-47](https://github.com/okijhhyu/element-tiptap-vue3/assets/75586002/3d8847fa-aea7-4fe6-aa2e-e54d51a8da18)

you should write your own button Link (AddlinkCommand component)
```Vue
<template>
  <div>
    <command-button
      :is-active="editor.isActive('link')"
      :readonly="isCodeViewMode"
      :command="openAddLinkDialog"
      :enable-tooltip="enableTooltip"
      :tooltip="t('editor.extensions.Link.add.tooltip')"
      icon="link"
      :button-icon="buttonIcon"
    />

    <el-dialog
      v-model="addLinkDialogVisible"
      :title="t('editor.extensions.Link.add.control.title')"
      :append-to-body="true"
      width="400px"
      class="el-tiptap-edit-link-dialog"
    >
      <el-form :model="linkAttrs" label-position="right" size="small">
        <el-form-item
          :label="t('editor.extensions.Link.add.control.href')"
          prop="href"
        >
          <el-input
            v-model="linkAttrs.href"
            autocomplete="off"
            :placeholder = placeholder />
        </el-form-item>

        <el-form-item prop="openInNewTab">
          <el-checkbox v-model="linkAttrs.openInNewTab">
            {{ t('editor.extensions.Link.add.control.open_in_new_tab') }}
          </el-checkbox>
        </el-form-item>
      </el-form>

      <template #footer>
        <el-button size="small" round @click="closeAddLinkDialog">
          {{ t('editor.extensions.Link.add.control.cancel') }}
        </el-button>

        <el-button
          type="primary"
          size="small"
          round
          @mousedown.prevent
          @click="addLink"
        >
          {{ t('editor.extensions.Link.add.control.confirm') }}
        </el-button>
      </template>
    </el-dialog>
  </div>
</template>

<script lang="ts">
import { defineComponent, inject } from 'vue';
import {
  ElDialog,
  ElForm,
  ElFormItem,
  ElInput,
  ElCheckbox,
  ElButton,
} from 'element-plus';
import { Editor } from '@tiptap/core';
import CommandButton from '../CommandButton.vue';

export default defineComponent({
  name: 'AddLinkCommandButton',

  components: {
    ElDialog,
    ElForm,
    ElFormItem,
    ElInput,
    ElCheckbox,
    ElButton,
    CommandButton,
  },

  props: {
    editor: {
      type: Editor,
      required: true,
    },
    buttonIcon: {
      default: '',
      type: String
    },
    placeholder: {
      default: '',
      type: String
    }
  },

  setup() {
    const t = inject('t');
    const enableTooltip = inject('enableTooltip', true);
    const isCodeViewMode = inject('isCodeViewMode', true);

    return { t, enableTooltip, isCodeViewMode };
  },

  data() {
    return {
      linkAttrs: {
        href: '',
        openInNewTab: true,
      },

      addLinkDialogVisible: false,
    };
  },

  watch: {
    addLinkDialogVisible() {
      this.linkAttrs = { href: '', openInNewTab: true };
    },
  },

  methods: {
    openAddLinkDialog() {
      this.addLinkDialogVisible = true;
    },

    closeAddLinkDialog() {
      this.addLinkDialogVisible = false;
    },

    addLink() {
      if (this.linkAttrs.openInNewTab) {
        this.editor.commands.setLink({
          href: this.linkAttrs.href,
          target: '_blank',
        });
      } else {
        this.editor.commands.setLink({ href: this.linkAttrs.href });
      }

      this.closeAddLinkDialog();
    },
  },
});
</script>

```  about editing text give me week. i will think about this, when complete problems with work

**Comment by zhangY94** [comment]
Thank you again for your help

**Comment by okijhhyu** [comment]
> Thank you again for your help

 No problem ask. If u  need help
